command/connect/envoy/envoy.go

```diff
@@ -191,6 +191,11 @@ func (c *cmd) templateArgs() (*BootstrapTplArgs, error) {
 	httpCfg := api.DefaultConfig()
 	c.http.MergeOntoConfig(httpCfg)
 
+	// Trigger the Client init to do any last-minute updates to the Config.
+	if _, err := api.NewClient(httpCfg); err != nil {
+		return nil, err
+	}
```

Comment:
This is a bit of a smell that we are using a client constructor to perform side effects in other places. I don't think it's something new you are introducing though - more that that constructor should probably not have been where a lot of that default setup code lives and so should be re-usable without this.

It's probably not the right time to go shaving that particular yak just now but maybe one day we can aspire to do better 😄.

Reply:
Yeah I have to monkey with all of this for rotation, and I remember the current wiring is less than awesome when you have tokens _updating_ in the client so it'll be addressed.